python-sdk/actionml/__init__.py

```diff
@@ -0,0 +1,634 @@
+        :param dataset_id:
+        :returns: AsyncRequest object.
+        """
+        request = AsyncRequest("GET", self._add_segment(dataset_id))
+        request.set_response_handler(self._ok_response_handler)
+        self._connection.make_request(request)
+        return request
+
+    def get(self, dataset_id: str):
+        return self.async_get(dataset_id).get_response()
+
+    def async_create(self, dataset_id: str = None):
+        """
+        Asynchronously create dataset.
+        :param dataset_id: 
+        :return: 
+        """
+        data = {}
+        if dataset_id is not None:
+            data['id'] = dataset_id
+
+        request = AsyncRequest("POST", self.path, **data)
+        request.set_response_handler(self._create_response_handler)
+        self._connection.make_request(request)
+        return request
+
+    def create(self, dataset_id: str = None):
+        return self.async_create(dataset_id).get_response()
+
+    def async_delete(self, dataset_id: str):
+        request = AsyncRequest("DELETE", self._add_segment(dataset_id))
+        request.set_response_handler(self._delete_response_handler)
+        self._connection.make_request(request)
+        return request
+
+    def delete(self, dataset_id: str):
+        return self.async_delete(dataset_id).get_response()
+
+
+class EngineClient(BaseClient):
+    def __init__(self, url: str = "http://localhost:8080", threads=1, qsize=0, timeout=5):
+        self.path = "/engines"
+        super(EngineClient, self).__init__(url, threads, qsize, timeout)
+
+    def async_get(self, engine_id: str):
+        """
+        Asynchronously get an engine info from PIO Kappa Server.
+        :param engine_id:
+        :returns: AsyncRequest object.
+        """
+        request = AsyncRequest("GET", self._add_segment(engine_id))
+        request.set_response_handler(self._ok_response_handler)
+        self._connection.make_request(request)
+        return request
+
+    def get(self, engine_id: str):
+        return self.async_get(engine_id).get_response()
+
+    def async_create(self, engine_id: str = None):
+        """
+        Asynchronously create engine.
+        :param engine_id: 
+        :return: 
+        """
+        data = {}
+        if engine_id is not None:
+            data['id'] = engine_id
+
+        request = AsyncRequest("POST", self.path, **data)
+        request.set_response_handler(self._create_response_handler)
+        self._connection.make_request(request)
+        return request
+
+    def create(self, engine_id: str = None):
+        return self.async_create(engine_id).get_response()
+
+    def async_delete(self, engine_id: str):
+        request = AsyncRequest("DELETE", self._add_segment(engine_id))
+        request.set_response_handler(self._delete_response_handler)
+        self._connection.make_request(request)
+        return request
+
+    def delete(self, engine_id: str):
+        return self.async_delete(engine_id).get_response()
+
```

Comment:
Similar comments to Dataset. Correct to have these, but only the methods required by the REST API or admin CLI may be implemented, the rest at "not implemented"